components/datetime/src/any/datetime.rs

```diff
@@ -248,18 +248,14 @@ impl DateTimeFormatter {
             + DataProvider<JapaneseExtendedErasV1Marker>
             + ?Sized,
     {
-        // TODO(#2188): Avoid cloning the DataLocale by passing the calendar
-        // separately into the raw formatter.
-        let mut locale_with_cal = locale.clone();
-
-        let calendar = AnyCalendar::try_new_for_locale_unstable(data_provider, &locale_with_cal)?;
+        let calendar = AnyCalendar::try_new_for_locale_unstable(data_provider, locale)?;
         let kind = calendar.kind();
-        kind.set_on_data_locale(&mut locale_with_cal);
 
         let patterns = PatternSelector::for_options(
             data_provider,
             calendar::load_lengths_for_any_calendar_kind(data_provider, locale, kind)?,
-            &locale_with_cal,
+            locale,
+            &kind.as_bcp47_value(),
```

Comment:
It doesn't matter too much since this is experimental code, but I could go either way on whether to pass the `Value` by value or by reference.

- Passing by reference means we need to `Clone` it, which happens in certain circumstances but not all the time (only when using components bag in a non-ethiopic locale).
- Passing by value means we need to `Drop` it, which happens every time.